purchases/src/test/java/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -3077,6 +3109,108 @@ class PurchasesTest {
         assertThat(capturedLambda).isNotNull
     }
 
+    @Test
+    fun `syncing an Amazon transaction without price nor currency code posts purchase data to backend`() {
+        purchases.finishTransactions = false
+
+        val skuParent = "sub"
+        val skuTerm = "sub.monthly"
+        val purchaseToken = "crazy_purchase_token"
+        val amazonUserID = "amazon_user_id"
+
+        every {
+            mockBillingAbstract.normalizePurchaseData(
+                productID = skuParent,
+                purchaseToken = purchaseToken,
+                storeUserID = amazonUserID,
+                captureLambda(),
+                any()
+            )
+        } answers {
+            lambda<(String) -> Unit>().captured.also {
+                it.invoke(skuTerm)
+            }
+        }
+
+        every {
+            mockCache.getPreviouslySentHashedTokens()
+        } returns setOf()
+
+        purchases.syncObserverModeAmazonPurchase(
+            productID = skuParent,
+            receiptId = purchaseToken,
+            amazonUserID = amazonUserID,
+            price = null,
+            isoCurrencyCode = null
+        )
+
+        val productInfo = ReceiptInfo(productID = skuTerm)
+        verify(exactly = 1) {
+            mockBackend.postReceiptData(
+                purchaseToken = purchaseToken,
+                appUserID = appUserId,
+                isRestore = false,
+                observerMode = true,
+                subscriberAttributes = emptyMap(),
+                receiptInfo = productInfo,
+                storeAppUserID = amazonUserID,
+                onSuccess = any(),
+                onError = any()
+            )
+        }
+    }
+
+    @Test
+    fun `syncing an Amazon transaction with zero price posts correct purchase data to backend`() {
+        purchases.finishTransactions = false
+
+        val skuParent = "sub"
+        val skuTerm = "sub.monthly"
+        val purchaseToken = "crazy_purchase_token"
+        val amazonUserID = "amazon_user_id"
+
+        every {
+            mockBillingAbstract.normalizePurchaseData(
+                productID = skuParent,
+                purchaseToken = purchaseToken,
+                storeUserID = amazonUserID,
+                captureLambda(),
+                any()
+            )
+        } answers {
+            lambda<(String) -> Unit>().captured.also {
+                it.invoke(skuTerm)
+            }
+        }
+
+        every {
+            mockCache.getPreviouslySentHashedTokens()
+        } returns setOf()
+
+        purchases.syncObserverModeAmazonPurchase(
+            productID = skuParent,
+            receiptId = purchaseToken,
+            amazonUserID = amazonUserID,
+            price = 0.0,
+            isoCurrencyCode = null
+        )
+
+        val productInfo = ReceiptInfo(productID = skuTerm)
```

Comment:
this would create ReceiptInfo with the default parameter values right? and wouldn't we expect them to be 0.0/null instead of the defaults?

Reply:
the defaults are actually null, but as you say I think it makes more sense to be explicit about it

Reply:
i thought the default for price was `productDetails?.priceAmountMicros?.div(MICROS_MULTIPLIER.toDouble())`?

Reply:
We are not passing productDetails, so it will be null in this case

Reply:
What I meant is that for this case, since productDetails is null, the defaults of currency and price are null too. Not that the defaults are always null. Sorry for the confusion. I added an explicit null so it's more clear